thrust/system/cuda/detail/core/triple_chevron_launch.h

```diff
@@ -55,57 +58,63 @@ namespace launcher {
           shared_mem(shared_mem_),
           stream(stream_) {}
 
-    template<class K, class... Args>
-    cudaError_t __host__
-    doit_host(K k, Args const&... args) const
+    template <class K, class... Args>
+    __host__
+    cudaError_t doit_host(K k, Args const&... args) const
     {
       k<<<grid, block, shared_mem, stream>>>(args...);
       return cudaPeekAtLastError();
     }
 
-    template<class T>
-    size_t __device__
-    align_up(size_t offset) const
+    template <class T>
+    __device__
+    std::size_t align_up(std::size_t offset) const
     {
-      size_t alignment = alignment_of<T>::value;
+      std::size_t alignment = alignment_of<T>::value;
       return alignment * ((offset + (alignment - 1))/ alignment);
     }
 
-    size_t __device__ argument_pack_size(size_t size) const { return size; }
+    __device__
+    std::size_t argument_pack_size(std::size_t size) const { return size; }
+
     template <class Arg, class... Args>
-    size_t __device__
-    argument_pack_size(size_t size, Arg const& arg, Args const&... args) const
+    __device__
+    std::size_t argument_pack_size(std::size_t size,
+                                   Arg const& arg,
+                                   Args const&... args) const
     {
       size = align_up<Arg>(size);
       return argument_pack_size(size + sizeof(Arg), args...);
     }
 
     template <class Arg>
-    size_t __device__ copy_arg(char* buffer, size_t offset, Arg arg) const
+    __device__
+    std::size_t copy_arg(char* buffer, std::size_t offset, Arg arg) const
     {
       offset = align_up<Arg>(offset);
-      for (int i = 0; i != sizeof(Arg); ++i)
-        buffer[offset+i] = *((char*)&arg + i);
+      for (int i = 0; i != static_cast<int>(sizeof(Arg)); ++i)
```

Comment:
Should we just use auto for the loop index instead of the static_cast?
```suggestion
      for (auto i = 0; i != sizeof(Arg); ++i)
```

Reply:
This is needed to fix a conversion warning, `auto` will have the same issue since it deduces `0` to `int`. The cast is what we want in this case.

Reply:
Yeah stupid me, that should be `size_t` instead of auto

Reply:
Nah, you're good -- it's a valid suggestion, but in this context we want to stick with `int`:

This code is executed on a CUDA device, and in this case it's best to avoid using 64-bit ints when 32-bits will suffice. Register space can be quite limited in some kernels, and these sorts of changes can have surprising impacts on performance.